### PR TITLE
Make app strict TypeScript + linting (and add a shared project)

### PR DESCRIPTION
**Pull request by TheCleric**
Sequel to #1223 . Now the whole repo should be strict!

### CODE REVIEW

**Review by ronjouch — CHANGES_REQUESTED** [review]

File: app/.eslintrc.js
Comment (TheCleric) [reply app/.eslintrc.js:16]: I thought about trying to make a base one that these import, but not sure how well that would work. What do you think?

File: app/package.json
Comment (TheCleric) [reply app/package.json:15]: Yep!

File: src/infer/inferIcon.ts
Comment (TheCleric) [reply src/infer/inferIcon.ts:6]: Yep! It's actually what TypeScript (or ESLint?) wanted. It did NOT work at first, but was resolved by setting `"esModuleInterop": true,` in the `tsconfig-base.json`

File: app/src/helpers/windowEvents.ts
Comment (TheCleric) [reply app/src/helpers/windowEvents.ts:10]: Yep. I did the following:
- `npm pack`
- `npm i -g ./nativefier-44.0.4.tgz`
- `nativefier https://google.com ~/Downloads`
- Verified the app launched.

I didn't see an exact recipe there, so let me know if this is insufficient

File: app/src/helpers/windowEvents.ts
Comment (TheCleric) [reply app/src/helpers/windowEvents.ts:5]:
```suggestion
```

File: src/.eslintrc.js
Comment (TheCleric) [reply src/.eslintrc.js:40]: Good point.

File: src/.eslintrc.js
Comment (TheCleric) [reply src/.eslintrc.js:2]: Yeah I replied to one of your other comments wondering that myself. I'll attempt it.

Comment (ronjouch) [reply src/.eslintrc.js:4]: @TheCleric okay. For me to understand: why the split into several `.eslintrc`s ?

File: app/src/helpers/windowEvents.ts
Comment (ronjouch) [reply app/src/helpers/windowEvents.ts:9]: Yeah that's it 👍

File: src/.eslintrc.js
Comment (TheCleric) [reply src/.eslintrc.js:4]: Because each has its own slightly different tsconfig.json since each is its own ts project and you have to reference one tsconfig per eslintrc.

Comment (ronjouch) [reply src/.eslintrc.js:4]: > you have to reference one tsconfig per eslintrc.

Thx, that's what I had missed.

I like what you did with sharing what can be shared, by accessing `base.{plugins, parser, extends, rules}`. Neat.

Comment (TheCleric) [reply src/.eslintrc.js:4]: Thanks!

**Comment by TheCleric** [comment]
Let me know if there's anything else you'd like done here @ronjouch :)

**Review by ronjouch — APPROVED** [review]
> Let me know if there's anything else you'd like done here

@TheCleric nope, all good.

1. Letting you merge this before/after https://github.com/nativefier/nativefier/pull/1235 as you want
2. Tell me if you want a release before you merge this. Right now, the only code commit since strict CLI (44.0.4) is https://github.com/nativefier/nativefier/pull/1229 .

**Comment by TheCleric** [comment]
> 1. Letting you merge this before/after https://github.com/nativefier/nativefier/pull/1235 as you want
> 
> 2. Tell me if you want a release before you merge this. Right now, the only code commit since strict CLI (44.0.4) is https://github.com/nativefier/nativefier/pull/1229 .

Let's merge 1235, cut a release, merge this, then cut another. As well there's a new Electron 12.

**Comment by ronjouch** [comment]
> Let's merge 1235, cut a release, merge this, then cut another. As well there's a new Electron 12.

@TheCleric I just released 44.0.5 with the recent tray/zoom/externalurl fixes, and an Electron bump. Letting you rebase & merge this whenever convenient, and I'll cut a release.

**Comment by TheCleric** [comment]
Merged!

**Comment by ronjouch** [comment]
> Merged!

:tada: ! Released as [44.0.6](https://github.com/nativefier/nativefier/releases/tag/v44.0.6).

**Pull request merged**
